Cython/Utility/FunctionArguments.c

```diff
@@ -201,7 +225,18 @@ static int __Pyx_ParseOptionalKeywords(
     PyObject*** name;
     PyObject*** first_kw_arg = argnames + num_pos_args;
 
-    while (PyDict_Next(kwds, &pos, &key, &value)) {
+    while (1) {
+        if (CYTHON_METH_FASTCALL && PyTuple_Check(kwds)) {
```

Comment:
I decided not to do that since the `PyTuple_Check(kwds)` call is extremely fast. Given that it's called on an already slow path (parsing keyword arguments), I figured that it wasn't worth the effort to optimize this better.

One thing I can do is doing this check just once instead of every loop iteration (the compiler doesn't know that the result of `PyTuple_Check(kwds)` won't change during the loop).